streams-tests/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -1405,7 +1403,7 @@ class StreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRu
     val s2 = Stream(1, 2)
 
     s1.mergeWith(s2)(_.toString, _.toString)
-      .run(Sink.succeed("done"))
+      .run(Sink.succeed[String, String]("done"))
```

Comment:
Arg. This is one of the places where it's a loss.

Reply:
@jdegoes Oh it's good to have the grandmaster of variance and inference here.

There's a really annoying issue that the Stream's  element type parameter  isn't used to infer the Sink's input type parameter - in `run`, `transduce` and others. Any thoughts on that?

Reply:
Can you paste signatures and I'll make it work (if it's possible 😆 )? Thank you! 🙏

Reply:
```scala
trait ZSink[-R, +E, A, +B]
object ZSink {
  final def succeed[A, B](b: B): ZSink[Any, Nothing, A, B] = new ZSink[Any, Nothing, A, B] {}
}

trait ZStream[-R, +E, +A] {
  def run[R1 <: R, E1 >: E, A1 >: A, B](sink: ZSink[R1, E1, A1, B]): Unit = ()
}

val stream = new ZStream[Any, Nothing, String] {}
stream.run(ZSink.succeed("OK"))

// cmd0.sc:11: type mismatch;
// found   : ammonite.$sess.cmd0.ZSink[Any,Nothing,Nothing,String]
// required: ammonite.$sess.cmd0.ZSink[Any,Nothing,String,String]
```

Reply:
@jdegoes ^

Reply:
Fixed in Dotty, at least

Reply:
I think this case is beyond help. The covariance on `A` in `ZStream` is interacting with the invariance on `ZSink` to cause the problem, and we certainly don't want to eliminate covariance on `ZStream`